m3u8/writer.go

```diff
@@ -73,6 +81,24 @@ func (p *MasterPlaylist) Encode() *bytes.Buffer {
 	if p.IndependentSegments() {
 		p.buf.WriteString("#EXT-X-INDEPENDENT-SEGMENTS\n")
 	}
+	if len(p.Defines) > 0 {
```

Comment:
I think we should skip this check, since the range function will handle the zero-length case.